### ISSUE TITLE
Banding artifacts against gradual sloping obstacles

### ISSUE THREAD

**Issue by rlguy** [rlguy/Blender-FLIP-Fluids-Beta#1]
- **CPU:** Intel i7 7700
- **RAM:** 32 GB
- **GPU:** GTX 1070, 8 GB
- **OS:** Windows 8.1
- **Blender Version:** 2.79
- **Addon Version:** 0.1.1

A banding effect can be seen in the fluid particles when a thin layer of fluid is sliding down a gradual sloping obstacle.

Screenshot:
![banding_artifacts](https://user-images.githubusercontent.com/4285191/35420815-78dda132-01f4-11e8-97c3-c3e4b602ce34.jpg)

Example .blend file:
https://www.dropbox.com/s/a9om9oimpbiii0a/gradual_slope_banding_artifacts.blend?dl=0

**Comment by EverSimo** [comment]
I've noticed this behavior too. 

[Beach Render (unfinished)](https://r4---sn-q4flrner.googlevideo.com/videoplayback?id=2aef60ac7caf24eb&itag=37&source=picasa&begin=0&requiressl=yes&pl=23&sc=yes&ei=QjWPWs-wH9Cz-wWL76K4Ag&susc=ph&app=fife&mime=video/mp4&cnr=14&lmt=1519330605857962&ip=170.84.59.140&ipbits=8&expire=1519341922&sparams=app,cnr,ei,expire,id,ip,ipbits,itag,lmt,mime,mm,mn,ms,mv,pl,requiressl,sc,source,susc&signature=4B5ECFC619220342EEA3F564E194F27A2DDEDE36.6CF328D07A7B2EECDE293C925FE36428217F24C2&key=cms1&redirect_counter=1&cm2rm=sn-bg0lk7z&req_id=dcbe6d0142cca3ee&cms_redirect=yes&mm=34&mn=sn-q4flrner&ms=ltu&mt=1519334557&mv=m)

Any hope to get a solution for this artifacts?

**Comment by rlguy** [comment]
Yes, I hope to get this fixed, but have not yet had time to test different solutions to this problem.

By the way, I am not able to view your video link:

```
Access to r4---sn-q4flrner.googlevideo.com was denied
You don't have authorisation to view this page.
HTTP ERROR 403
```

**Comment by EverSimo** [comment]
Please, try again... sorry...

[Beach Render (unfinished)](https://photos.app.goo.gl/rqIq0teehcemB1J33)

**Comment by rlguy** [comment]
Hey niklassjogren,

This looks like it is actually a meshing issue rather than a simulation banding issue.

The mesh generator uses a grid to convert particles to a triangle mesh. When thin layers of fluid are against an obstacle, the particles may not show up well on the meshing grid. Some fixes for this are to:

1. **Increase surface subdivision level** - This will create a finer mesh grid and generate a higher quality fluid mesh. This will also take more time to compute. This setting is located in the FLIP Fluid Surface settings.
2. **Increase particle scale** - This will increase particle size when meshing. The particles will show up better on the meshing grid. This does not take much more time to compute, but your fluid surface may end up looking 'blobby'. Try setting the scale to something like 1.50. This setting is also located in the FLIP Fluid Surface settings.

**Comment by RemiGraphics** [comment]
![image](https://user-images.githubusercontent.com/37308267/38655112-57e6e978-3dd8-11e8-87e6-35bd3180f6e5.png)
Seeing similar issues in my wave simulations. Hoping this issue hasn't been forgotten :)

**Comment by rlguy** [comment]
This issue haunts me and I can assure you it has not been forgotten! I spent a few hours earlier this week to try a few things, but did not find a solution. I've come to the conclusion that I should build some debugging tools to better visualise/understand what could be causing this issue whether it be a bug or flaw in the simulation method. I've actually planned to devote the entire week of April 23rd - 27th to work on this if needed!

Anyways, it's possible that the visual artifacts in your mesh could be improved with the tips in the [comment above yours](https://github.com/rlguy/Blender-FLIP-Fluids-Beta/issues/1#issuecomment-368628770). It's also possible that it could make the simulation banding artifacts **more** noticable.

**Comment by enzyme69** [comment]
I am getting same issue with what I am doing (obstacle with lots of curvey surface). Banding happens, when radius increased, it becomes too blobbies...

**Comment by rlguy** [comment]
Hey @enzyme69, if increasing particle radius is making the fluid too blobby, I would suggest trying to increase the subdivision level to 1. This will improve the detail of the surface and can help fill in the banding gaps.

**Comment by PavelBlend** [comment]
Another way to get rid of these artifacts is to disable the "Mesh Around Obstacles" checkbox. But in the last experimental build there is no this parameter. I often turned off this checkbox.

**Comment by rlguy** [comment]
Meshing in the later versions have changed and the _Mesh Around Obstacles_ option was removed due to being out of date and no longer compatible. The new default setting for how fluid is mesh around obstacles (Mesh Inside Obstacle) works similar to the old _Mesh Around Obstacles_ and can help reduce meshing artifacts.

Note: this issue is about simulation banding artifacts rather than mesh banding artifacts. The simulation banding artifacts cause particles to group and align to the grid:

![artifact_grouping](https://user-images.githubusercontent.com/4285191/58124841-ff164700-7bc3-11e9-8126-7c0327e1ecac.jpg)

Many of the comments in this thread are related to mesh banding artifacts ([of which here is a fix](https://github.com/rlguy/Blender-FLIP-Fluids/wiki/Scene-Troubleshooting#mesh-banding-artifacts-against-curved-obstacles)). I'll add a note about this to the top post and hide some comments to make this issue more clear.

**Comment by zx-82** [comment]
Any updates on this ?
I am developping my own fluid simulator and am confronted with the same issue. I suspect that it is related to the "small cell" problem (common to all cut cell based liquid solvers) because along a gradual slope there will be cells (fractions) that only contain an arbitrary tiny amount of fluid. In engineering CFD codes this problem is addressed by merging "tiny cells"  with neighboring cells. While solvers used in computer graphics seem to rely on Bridson's simple "clamping trick". But this will also leave some residual "stair casing", I'm afraid.

**Comment by rlguy** [comment]
There is no update on this issue and it has been at least two years since I have had a chance to look further into the problem. The "small cell" problem sounds like a good lead for something to look into for this issue. Thank you for the tip, but I am sorry to say that I do not have any advice at the moment.

Good luck with your fluid simulator development!

**Comment by zx-82** [comment]
Hi, 

 

>I am sorry to say that I do not have any advice at the moment.

 

Thanks for your reply which is actually helpful because your (and mine so far) failure to find a simple way to resolve it, indicates that this banding is indeed related to math method/algorithm. So at least no need to wasting time chasing an imaginary bug in the software.

 

Best Regards,

  Bernhard

 
 

Gesendet: Montag, 01. November 2021 um 18:46 Uhr
Von: "Ryan Guy" ***@***.***>
An: "rlguy/Blender-FLIP-Fluids" ***@***.***>
Cc: "zx-82" ***@***.***>, "Comment" ***@***.***>
Betreff: Re: [rlguy/Blender-FLIP-Fluids] Banding artifacts against gradual sloping obstacles (#1)


 

There is no update on this issue and it has been at least two years since I have had a chance to look further into the problem. The "small cell" problem sounds like a good lead for something to look into for this issue. Thank you for the tip, but I am sorry to say that I do not have any advice at the moment.

Good luck with your fluid simulator development!

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
